sql/src/main/java/io/crate/metadata/settings/CrateSettings.java

```diff
@@ -287,11 +288,46 @@ Settings settings() {
     private Map<String, ReferenceImplementation> buildReferenceTree() {
         Map<String, ReferenceImplementation> referenceMap = new HashMap<>(BUILT_IN_SETTINGS.size());
         for (CrateSetting crateSetting : BUILT_IN_SETTINGS) {
+            if (crateSetting.isGroupSetting()) {
+                Map<String, Settings> settingsMap = initialSettings.getGroups(crateSetting.getKey(), true);
+                for (Map.Entry<String, Settings> entry : settingsMap.entrySet()) {
+                    buildGroupSettingReferenceTree(crateSetting.getKey(), entry, referenceMap);
+                }
+            }
             buildReferenceTree(referenceMap, crateSetting);
         }
         return referenceMap;
     }
 
+    @VisibleForTesting
+    void buildGroupSettingReferenceTree(String prefix, Map.Entry<String, Settings> settingEntry, Map<String, ReferenceImplementation> referenceMap) {
```

Comment:
pls respect max line width, also pls pass in the key and value instead of a map.entry